src/compiler/crystal/exception.cr

```diff
@@ -242,22 +242,27 @@ module Crystal
 
     def minimize_indentation(source)
       min_leading_white_space =
-        source.map do |line|
-          if match = line.match(/^(\s+)\S/)
-            spaces = match[1]?
-            spaces.size if spaces
-          end
-        end
-          .compact
-          .min
+        source.min_of? { |line| leading_white_space(line) } || 0
 
-      source = source.map do |line|
-        replace_leading_tabs_with_spaces(line).lchop(" " * min_leading_white_space)
+      if min_leading_white_space > 0
+        source = source.map do |line|
+          replace_leading_tabs_with_spaces(line).lchop(" " * min_leading_white_space)
+        end
       end
 
       {source, min_leading_white_space}
     end
 
+    private def leading_white_space(line)
+      match = line.match(/^(\s+)\S/)
+      return 0 unless match
+
+      spaces = match[1]?
+      return 0 unless spaces
+
+      spaces.size
```

Comment:
Since this part is refactored, we can simplify it by avoiding using a regex:
```cr
line.each_char_with_index do |char, index|
  return index if char != ' '
end
0
```

Reply:
No, `\s` matches tab, spaces, maybe also unicode space, I don't know.

This code runs once when the compiler finds an error, optimizing here has no impact at all. Maybe it's 5ms faster, the user can't possibly tell.

I also prefer avoiding changing code if it's not broken.

Reply:
Ho, there is `char.ascii_whitespace?`

But that's basically what's done here, the code is already changed.